keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ServerGroup.kt

```diff
@@ -47,7 +47,16 @@ data class ServerGroup(
   @JsonIgnore
   @get:ObjectDiffProperty(inclusion = EXCLUDED)
   val buildInfo: BuildInfo? = null
-)
+) {
+  init {
+    require(
+      capacity.desired != null && !scaling.hasScalingPolicies() ||
```

Comment:
I think you still want a desired value when spinning up a new ASG; otherwise, you'll have to wait a while for the scaling policies to kick in and adjust the number of instances, right?

Reply:
Would you be ok with a new `initialDesired` property that's only applicable when a cluster with scaling policies is deployed for the very first time?

Reply:
Although, if a user is declaring a new cluster that has never been deployed, why not set min to the intended starting size?

Reply:
I think I'd rather use the min as the initial size instead of having a new property.

Reply:
If it’s a new cluster is it really an issue that there may be a short wait before the scaling policies kick in?

Reply:
I would ask a few users of scaling policies right now what they'd want or expect. Since we don't use them on our team, I don't feel like we have enough information or context on what people will want here.